doc/dev/adr/0020-error-modeling-headers.md

```diff
@@ -0,0 +1,103 @@
+# ADR 19: Modeling User Errors
+
+## Context
+
+We have received feedback from users that they would like to have RPC responses communicate application level failures in the headers of the response rather than the payload of the response. 
+
+This feedback is, in part, because some applications want to route RPC responses without reading/deserializing the entire payload. It is also partly because [some applications cannot extend or change the payload model to accommodate error handling.](https://github.com/Azure/iot-operations-sdks/issues/488#issuecomment-2707496996).
+
+## Decision
+
+Our SDKs will add APIs on the command executor side that mark an RPC response as an "application error" by attaching two new MQTT user properties ("AppErrCode" and "AppErrPayload") whose values are a user-defined string and a user-defined payload object (serialized to bytes and then encoded as a UTF8 string) respectively. When a user wants to indicate that an RPC call failed with an application error they must provide the "AppErrCode" value, but "AppErrPayload" is optional.
```

Comment:
Why enforcing a user-defined payload object, serialized to bytes and then encoded as UTF8, to support code gen?

Reply:
Yes. We eventually want DTDL to model these headers so that we can provide strong typing to users via codegen